dotcom-rendering/src/web/components/Card/Card.stories.tsx

```diff
@@ -84,32 +86,26 @@ const CardsWithDifferentThemes = ({
 	design: ArticleDesign;
 	title: string;
 }) => {
-	const cards = [];
-	for (const [pillarName, pillarValue] of Object.entries(ArticlePillar)) {
-		// We need to check if numeric here because the values are also exported as keynames
-		// See: https://www.typescriptlang.org/play?#code/KYOwrgtgBAggTgFwJYGMA2wAKS1oIZxQDeAsAFBSVQBywA7gM5QC8UADADTlVQDyADkhBIA9iBZQAjFwpUAyvxGIJAJhk8AwmDQIwcYBIDM6qgBkkAM2AMEATwwSALDIC+5cijEMRGAHRoRAHMACl4AIwArYBQEX1AEOCRrYPhkdCwcfDgASmyAbnIAegAqcgBtbioygCI2ao4oatpGaoBdE0oayXrGgSFREDaOqBqVHuqFJQQhys7qw3GtHT1gGdk5x3HzKxt7VfbZkab6Bh62A-WjvuExHskLnhrJxB6VB6rqpd19HsN3ue21jsGHqjla5FaUHIxUKQA
-		if (Number.isNaN(Number(pillarName))) {
-			cards.push({
-				title: pillarName,
-				format: {
-					display,
-					design,
-					theme: pillarValue as ArticlePillar,
-				},
-			});
-		}
+	const cards: Array<{ title: string; format: ArticleFormat }> = [];
```

Comment:
I really need to scratch my itch and ask - why don't we use `map`?

Reply:
I am as puzzled as you are, but I think that should live in a subsequent PR.

Maybe the issue with the enum meant that it was hard to filter? Maybe the team is more comfortable with imperative approaches than purely functional ones?

Reply:
We use this pattern all over the codebase. If you search for ` = []` outside of the test files, you'll see what I mean (GitHub search isn't good for this one, so can't share a link).

![Screenshot 2022-05-27 at 11 56 13](https://user-images.githubusercontent.com/31692/170686629-e0295805-9e9d-4cbe-810d-2bc8dc8fade0.png)

Reply:
Yes, I don’t think it’s an ideal pattern. In many of these cases, it would be trivial to have a `map`, especially the enhancers.

[Github CodeSearch gives over 100 matches.](https://cs.github.com/guardian/dotcom-rendering?q=%3D+%5B%5D)

Reply:
There are `map`s used further down this file. It looks like, from the sample size of this one file, that `for...of...push` is being used when there's a need to filter, and `map` is being used when there isn't?

Reply:
Is this maybe a bit? 😉 
<img width="450" alt="Screenshot 2022-05-27 at 13 28 32" src="https://user-images.githubusercontent.com/1336821/170699306-e36efed5-1cd6-4234-99e1-d78b5e35673f.png">